src/Helpers/Normalizer.php

```diff
@@ -87,8 +87,12 @@ public static function normalize(mixed $definition, ?string $class = null): Defi
             return ArrayDefinition::fromConfig($config);
         }
 
+        if ($definition instanceof DefinitionInterface) {
```

Comment:
Move this check to first place in this method and remove check on instance of `ReferenceInterface`, because it extends `DefinitionInterface`. Also fix phpdoc according to this change.